src/socket_can_receiver_node.cpp

```diff
@@ -110,15 +113,19 @@ void SocketCanReceiverNode::receive()
     }
 
     try {
-      receive_id = receiver_->receive(frame_msg.data.data(), interval_ns_);
+      std::tie(receive_id, bus_time) = receiver_->receive(frame_msg.data.data(), interval_ns_);
     } catch (const std::exception & ex) {
       RCLCPP_WARN_THROTTLE(
         this->get_logger(), *this->get_clock(), 1000,
         "Error receiving CAN message: %s - %s",
         interface_.c_str(), ex.what());
       continue;
     }
-    frame_msg.header.stamp = this->now();
+    if (use_bus_time_) {
+      frame_msg.header.stamp = rclcpp::Time(static_cast<int64_t>(bus_time*1000U)) ;
```

Comment:
If you used a chrono duration type, this would be a duration cast. As an aside, I'm surprised `rclcpp::Time:Time` doesn't accept chrono but only integers.